README.rst

```diff
@@ -0,0 +1,323 @@
+
+Xen guest domains configuration
+*******************************
+
+The **zephyr-dom0-xt** application uses Xen guest domains configuration in `zephyr-xenlib`_
+domain configuration format defined by ``struct xen_domain_cfg``.
+
+The number of Xen guest domains configurations supported by **zephyr-dom0-xt** is specified
+by Kconfig option **CONFIG_DOM_CFG_NUMBER** (default 1).
+
+.. note::
+
+    Max number of supported Xen domains configuration now is 2 due to inability to integrate
+    Xen guest domains binary image files for more than 2 Xen domain configurations.
+
+Each Xen guest domains configurations shell be defined as global variable named **domd_cfg_X**,
+where X is in [0, CONFIG_DOM_CFG_NUMBER].
+
+One configuration example, ``CONFIG_DOM_CFG_NUMBER=1``:
+
+.. code:: C
+
+    struct xen_domain_cfg domd_cfg_0 = {
+    }
+
+Two configurations example, ``CONFIG_DOM_CFG_NUMBER=2``:
+
+.. code:: C
+
+    struct xen_domain_cfg domd_cfg_0 = {
+    }
+
+    struct xen_domain_cfg domd_cfg_1 = {
+    }
+
+There is an assumption that if some Xen quest domain intend to be defined as
+``Xen drivers domain (DomD)``, its configuration shall be defined as ``domd_cfg_0``.
+
+The **zephyr-dom0-xt** selects Xen guest domains configurations file (containing ``domd_cfg_X``)
+basing on:
+
+- The **CONFIG_USE_DEFAULT_DOM_CFG** (default=y) Kconfig option. if selected the ``src/domd_cfg.c``
+  file will be used (backward compatible behavior).
+- if CONFIG_USE_DEFAULT_DOM_CFG=n the Xen guest domains configurations file will be selected basing
+  on Zephyr board and **CONFIG_DOM_CFG_BOARD_EXT** Kconfig option as following
+
+.. code-block:: text
+
+      if DEFINED(CONFIG_DOM_CFG_BOARD_EXT)
+        use  src/dom_cfg/${BOARD}_${CONFIG_DOM_CFG_BOARD_EXT}.c
+      else()
+        use src/dom_cfg/${BOARD}.c
+
+For example Zephyr board is **rpi_5** and `CONFIG_USE_DEFAULT_DOM_CFG=n``:
+
+.. code-block:: text
+
+    src/dom_cfg/rpi_5.c
+    CONFIG_DOM_CFG_BOARD_EXT=\"domd\" ---> src/dom_cfg/rpi_5_domd.c
+
+Xen guest domains binary images
+*******************************
+
+The **zephyr-dom0-xt** takes pre-built Xen guest domains images from ``prebuilt/`` folder and
+they are integrated directly into Zephyr **zephyr-dom0-xt**. Now only two sets of Xen guest images
+is supported which names are defined by following Kconfig options:
+
+- **CONFIG_DOM_CFG_DOMD_IMAGE_BIN_FILE**: Location for Domain-D kernel binary.
+  For RPI 5 it is ``zephyr_blinky.bin`` corresponding to Zephyr ``samples/basic/blinky`` sample.
+- **CONFIG_DOM_CFG_DOMD_DTB_BIN_FILE**: Location for Domain-D device-tree binary.
+- **CONFIG_DOM_CFG_DOMU1_IMAGE_BIN_FILE**: Location for Domain-U kernel binary
+  For RPI 5 it is ``zephyr_sync.bin`` corresponding to Zephyr ``samples/synchronization`` sample.
+- **CONFIG_DOM_CFG_DOMU1_DTB_BIN_FILE**: "Location for Domain-U device-tree"
+
+To use custom binary images they shell be copied into ``prebuilt/`` folder.
```

Comment:
again, this is not mandatory when you have Kconfig option with name, it can be easily converted to path